### ISSUE TITLE
Use bat with a light coloured background?

### ISSUE THREAD

**Issue by girgen** [sharkdp/bat#3116]
Hi!

Is there a way to negate the colours (so white is black, eg)? I use a bone white background for better contrast, and bat isn't really usable out of the box then. Perhaps there's a switch for it, but I didn't see one?

Thanks!

**Comment by keith-hall** [comment]
You can use a different theme: https://github.com/sharkdp/bat?tab=readme-ov-file#highlighting-theme.
https://github.com/sharkdp/bat/pull/2896 has recently been merged which should improve theme selection by default.

**Comment by janxkoci** [comment]
For me the syntax highlighting is visible on white background, albeit higher contrast would improve it for sure. But with plain `txt` files I get white text on white background, which is pretty bad - it took me a while to realize `bat` was even working.

Using `bat 0.24.0 (9cecc21-modified)` installed with `conda` from conda-forge channel, on Manjaro GNOME (default shell is zsh, if that matters).

**Comment by janxkoci** [comment]
Ok weird, I've installed v0.25 on another machine and I've got the same issue again 🤔

**Comment by skullydazed** [comment]
It would be really good if you used default colors visible on both white and black backgrounds. I didn't set out to use bat, but paru uses bat, and they're pointing at bat as the reason I can't view my diff.

**Comment by janxkoci** [comment]
I don't believe there is one set of colours that would work well on both dark and light backgrounds, with enough contrast and readability. Not to mention people using themes etc. 

Bat obviously needs to detect the current theme (light or dark at minimum) and use the appropriate colours. I think bat is already meant to do that but it seems broken at the moment.

PS: bat version 0.26 seems to work fine (changing colours based on terminal theme) on the machine I have with me at the moment, running LMDE7 Cinnamon and tested with GNOME terminal & Tilix. Bat is from conda-forge.

**Comment by skullydazed** [comment]
Having one set of colors that works well is a challenge, but the current situation is a set of colors that don't work at all. Having colors that are simply too low contrast is better than having foreground and background the exact same color.

**Comment by janxkoci** [comment]
What I'm saying is that the issue is somewhere else than with colours - I had the same version of bat on two machines producing different results. It's more fruitful to figure out what's going on and fix it properly, rather than hardcoding a fixed colour palette like cavemen, imho. This is Rust after all, it's supposed to be modern and all...

PS: for context, the actual problem can easily be in one of at least [four components](https://jvns.ca/blog/2024/11/26/terminal-rules/), or - more likely - their interplay.

**Comment by skullydazed** [comment]
We are each talking about different aspects of the same issue here, and both of our points can be addressed.

1. It should be possible in most circumstances to [detect the colors in use](https://stackoverflow.com/questions/50027162/determine-terminal-tty-background-color-at-runtime) and pick a sensible theme
2. When detection is not possible, the default theme should always set both fg and bg colors so that the text is visible, even if it doesn't match the user's preferences.

**Comment by janxkoci** [comment]
I just don't think it's possible to do #2 reliably (and accessibly), while I think doing #1 in 2025 should be straightforward - most desktops just give you that info. Heck, it even works in ssh sessions for me, usually..

Except when it doesn't work, which I'd like to figure out here..

**Comment by skullydazed** [comment]
When you find yourself in 2, you're right that you can't do it universally "right" and "accessible". But you can do "reasonable in most circumstances" by selecting a default that sets both fg and bg colors in all places. Even if it goes against someone's preference, or accessibility is reduced due to low contrast, it's still preferable to the current situation where a user using black-on-white terminals (which is the default for most terminal programs) can't read text.

In any case, I've shared my perspective and I think it's clear enough. Thanks for the dialog and hope you have a great new year!
